portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,33 +33,55 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
     public List<String> messages;
+    public List<String> oldComments;
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        response.setContentType("text/html;");
-        response.getWriter().println("Hello Ciara!");
+        Query query = new Query("comment").addSort("timestamp", SortDirection.DESCENDING);
+    
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
 
-        String json = convertToJsonUsingGson(messages);
+        oldComments = new ArrayList<>();
+        for (Entity entity : results.asIterable()) {
+            String comment = (String) entity.getProperty("comment");
+    
+            oldComments.add(comment);
+        }
 
-        response.setContentType("application/json;");
-        response.getWriter().println(messages);
-    }
-
-    @Override
-    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        String comment = request.getParameter("box");
-        messages.add(comment);
+        Gson gson = new Gson();
 
-        response.sendRedirect("/index.html");
+        response.setContentType("application/json;");
+        response.getWriter().println(gson.toJson(oldComments));
     }
 
     public void init(){
         messages = new ArrayList<>();
     }
 
-    private String convertToJsonUsingGson(List<String> messages) {
+    private String convertToJsonUsingGson(Query query) {
         Gson gson = new Gson();
```

Comment:
Does this code work?  The argument is changed to "Query query".  But query is not used. Instead the code is return gson.ToJson(messages).

Reply:
nvm, looks like you have deleted this method already.